feature: add login, register, hopper admin pages and debug button for VSense

SasiKumar-V-optimaise opened the pull request:
feature: add login, register, hopper admin pages and debug button for VSense

- Added LoginPage with Streamlit UI and authentication via AuthService
- Added RegisterPage for admin user creation with proper validation
- Added HopperAdminPage to view and update material → hopper mappings
- Implemented debug button in VSense settings to toggle OPTIM_STEPS
- Applied consistent UI styling and session state management

SasiKumar-V-optimaise committed:
feat: add login, register, hopper admin pages and debug button for VSense

- Added LoginPage with Streamlit UI and authentication via AuthService
- Added RegisterPage for admin user creation with proper validation
- Added HopperAdminPage to view and update material → hopper mappings
- Implemented debug button in VSense settings to toggle OPTIM_STEPS
- Applied consistent UI styling and session state management

SasiKumar-V-optimaise committed:
feat: add login, register, hopper admin pages and debug button for VSense

- Added LoginPage with Streamlit UI and authentication via AuthService
- Added RegisterPage for admin user creation with proper validation
- Added HopperAdminPage to view and update material → hopper mappings
- Implemented debug button in VSense settings to toggle OPTIM_STEPS
- Applied consistent UI styling and session state management

saikumar-voptimai commented on src/data/db.py:
```diff
@@ -0,0 +1,230 @@
+        except IntegrityError:
+            raise ValueError("Username already exists.")
+
+    def validate_user(self, username: str, password: str):
```
Specify the output type to these functions. This is how the functions should look like:

def validate_user(self, username: str, password: str) -> Tuple:

SasiKumar-V-optimaise committed:
fix: patch issues after adding Login, Register, and Hopper Admin pages

- improved debug button behavior for OPTIM_STEPS
- refined UI consistency across pages
